source/extensions/tracers/opentelemetry/trace_exporter.h

```diff
@@ -29,6 +29,24 @@ class OpenTelemetryTraceExporter : public Logger::Loggable<Logger::Id::tracing>
    * @return false When sending the request failed.
    */
   virtual bool log(const ExportTraceServiceRequest& request) = 0;
+
+  /**
+   * @brief Logs as debug the number of exported spans.
+   *
+   * @param request The protobuf-encoded OTLP trace request.
+   */
+  void logExportedSpans(const ExportTraceServiceRequest& request) {
+    if (!ENVOY_LOG_COMP_LEVEL_FINE_GRAIN_IF(ENVOY_LOGGER(), debug)) {
+      return;
+    }
```

Comment:
is this necessary?

Reply:
I wanted to avoid having to do all the ifs and checking in case debug log was not enabled. Not sure if there's a easier way to do that. Otherwise we will do it all the time, even though the log is not printed.

Reply:
I think the overhead of `resource_spans(0)` should be pretty trival and doesn't deserve to introduce additional code complexity.
But if you prefer it, please add some comments to explain it for future reader.

Reply:
Sounds good, I wasn't sure as well, but I agree with you. Removed!